internal/state/conditions/httproute.go

```diff
@@ -0,0 +1,72 @@
+package conditions
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+// RouteCondition defines a condition to be reported in the status of an HTTPRoute.
+type RouteCondition struct {
```

Comment:
I'm unclear why we have this struct and not use the imported type.

Reply:
This type includes the fields necessary to capture information about a condition in an HTTPRoute. This includes using Route-specific types from the Gateway API.

If you're thinking  metav1. Condition, that type has more fields that we need here and they are also not Route-specific.

Additionally, making a specialized condition type will statically prevent using it for other resources like Gateway and GatewayClass (once we start implementing their statuses).

Reply:
How effective is the typing though? If each must be converted back to a metav1.Condition in the end the specific types have limited value for the cost of extra API surface, and testing.

Isn't this also still possible?
`Gateway.Spec.Conditions = convertRouteConditions(routeConditions)`

Reply:
> Isn't this also still possible?

yep, but this is in a different package and only in one place when we need to convert conditions. 
Note that in the `state` package we capture all conditions and store them in the fields of the graph. Right now, we capture only a few of them. However, the number of conditions (conditions and reasons) will grow.

In my view, because the conditions for different resources are different and non-overlapping, better to put them in a separate type.